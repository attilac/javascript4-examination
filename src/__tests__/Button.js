import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button';


describe('button unit tests', ()=> {
  const children = <p />;
  const onClick = jest.fn();
  const defaultWrapper = shallow(<Button onClick={onClick}>{children}</Button>);
  const dangerWrapper = shallow(<Button danger onClick={onClick}>{children}</Button>);

  it('should call onClick', ()=> {
    defaultWrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should render default button', ()=> {
    expect(defaultWrapper.find('button').hasClass('bg-indigo-dark')).toBeTruthy();
  })

  it('should match default btn snapshot', ()=> {
    expect(defaultWrapper).toMatchSnapshot();
  })

  it('should render danger button', ()=> {
    expect(dangerWrapper.find('button').hasClass('bg-red-dark')).toBeTruthy();
  });

  it('should match danger btn snapshot', ()=> {
    expect(dangerWrapper).toMatchSnapshot();
  })
})