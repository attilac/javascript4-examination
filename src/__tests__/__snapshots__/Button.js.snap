// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button unit tests should match danger btn snapshot 1`] = `
<button
  className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
>
  <p />
</button>
`;

exports[`button unit tests should match default btn snapshot 1`] = `
<button
  className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full "
  data-test="button"
  onClick={[Function]}
>
  <p />
</button>
`;
